nifi-nar-bundles/nifi-hive-bundle/nifi-hive-processors/src/test/java/org/apache/nifi/processors/hive/TestConvertAvroToORC.java

```diff
@@ -320,6 +322,78 @@ public void test_onTrigger_complex_record() throws Exception {
         assertEquals(sampleBigDecimal, ((HiveDecimalWritable) decimalFieldObject).getHiveDecimal().bigDecimalValue());
     }
 
+    @Test
+    public void test_onTrigger_complex_records_with_bigdecimals() throws Exception {
+
+        Map<String, Double> mapData1 = new TreeMap<String, Double>() {{
+            put("key1", 1.0);
+            put("key2", 2.0);
+        }};
+
+        DatumWriter<GenericData.Record> writer;
+        DataFileWriter<GenericData.Record> fileWriter = null;
+        ByteArrayOutputStream out = new ByteArrayOutputStream();
+
+        Random random = new Random();
+        double bigDecimalUpperRange = 200;
+        for (int i = 0; i < 1000; ++i) {
+            ByteBuffer bigDecimalAsBytes = toByteBuffer(new BigDecimal(random.nextDouble() * bigDecimalUpperRange).setScale(2, RoundingMode.HALF_UP));
+            GenericData.Record record = TestNiFiOrcUtils.buildComplexAvroRecord(null, mapData1, "XYZ", 4L, Arrays.asList(100, 200), bigDecimalAsBytes);
+            if (i == 0) {
+                writer = new GenericDatumWriter<>(record.getSchema());
+                fileWriter = new DataFileWriter<>(writer);
+                fileWriter.create(record.getSchema(), out);
+            }
+            fileWriter.append(record);
+        }
```

Comment:
Generating 1000 records is not necessary. With this approach, there is still chance - although very little - that the values that are generated do not force the issue to present itself.
The issue appears, when the byte representation of the decimal value of the currently processed record occupies fewer bytes than that of the previous record. In this case memory garbage from the previous record's value are interpreted as part of the current record's field value. Thus, it is enough to have a few records in the right order to force the issue to appear. For example (let's assume the schema is (id (string), decVal (decimal)):
Record1: "A1", 3500.0
Record2: "B2", 0.0
Record3: "C3", 46000.0
Record4: "D4", 22.0
(Actually, the first 2 records of this example are enough.)
I would like to ask you to consider creating the test data this way. It would make the test code easier to understand, the execution to be more efficient (fewer records to work with) and the value assertion to be more precise (see my comment below).

Reply:
Thanks, I simplified the test